airbyte-integrations/connectors/source-s3/source_s3/s3file.py

```diff
@@ -43,6 +46,10 @@ def use_aws_account(provider: Mapping[str, str]) -> bool:
         aws_secret_access_key = provider.get("aws_secret_access_key")
         return True if (aws_access_key_id is not None and aws_secret_access_key is not None) else False
 
+    @staticmethod
+    def use_aws_default_credential_provider_chain(provider: Mapping[str, str]) -> bool:
```

Comment:
What do you think of refactoring this a bit: 
* create a `AuthenticationMethod` enum 
* create a method that returns the authentication method to use (`get_authentication_method`)
* use `get_authentication_method` in `_setup_boto_session` to create session according to the authentication method to use

eg: 
```python 
class AuthenticationMethod(Enum):
  ACCESS_KEY_SECRET_ACCESS_KEY = 1
  DEFAULT = 2
  
class S3File:
  def __init__(self, *args: Any, **kwargs: Any):
    super().__init__(*args, **kwargs)
    self.authentication_method = self.get_authentication_method()
    self._setup_boto_session()

  def get_authentication_method(self):
    if self._provider.get("aws_access_key_id") and self._provider.get("aws_secret_access_key"):
      return AuthenticationMethod.ACCESS_KEY_SECRET_ACCESS_KEY
    elif self._provider.get("use_aws_default_credential_provider_chain"):
      return AuthenticationMethod.DEFAULT
    else:
      raise Exception("Could not determine an authentication method according to current provider.")

  def _setup_boto_session(self):
    if self.authentication_method == AuthenticationMethod.ACCESS_KEY_SECRET_ACCESS_KEY:
      ...
    if self.authentication_method == AuthenticationMethod.DEFAULT:
      ...
```

Reply:
I'd be great if we could define at a single spot the specifics related to an authentication method (sessions and params definition) to make the rest of the implementation independent from the authentication method used. My point is to avoid several if / else statements on `use_aws_account` or `use_aws_default_credential_provider_chain`. It will improve maintainability  and eventual iterations on authentication methods.